src/transport_picoquic.cpp

```diff
@@ -466,8 +459,9 @@ PicoQuicTransport::PicoQuicTransport(const TransportRemote& server,
                                      const TransportConfig& tcfg,
                                      TransportDelegate& delegate,
                                      bool _is_server_mode,
-                                     LogHandler& logger)
-  : logger(logger)
+                                     const cantina::LoggerPointer& logger)
+  : logger(std::make_shared<cantina::Logger>("QUIC", logger))
```

Comment:
So each class will call `make_shared()` instead of assignment of logger to plus ref count?

Reply:
Our choice.  We can make a copy of the logger pointer or create a new pointer with a parent-child relationship (as is the case here). The benefit of the parent-child relationship is that each module will have its own module name (here `QUIC`) that appears in the logs.  Each of the parent logging object prepend their module names, too, making it easy to identify exactly what object emitted the log message.